frb_codegen/src/generator/rust/mod.rs

```diff
@@ -358,15 +359,20 @@ impl<'a> Generator<'a> {
 
         let (handler_func_name, return_type, code_closure) = match func.mode {
             IrFuncMode::Sync => (
-                "wrap_sync",
+                String::from("wrap_sync"),
                 Some("support::WireSyncReturn"),
                 format!(
                     "{code_wire2api}
                     {code_call_inner_func_result}"
                 ),
             ),
-            IrFuncMode::Normal | IrFuncMode::Stream { .. } => (
-                "wrap",
+            IrFuncMode::Normal => (
+                format!("wrap::<_,_,_,{}>", func.output.intodart_type(ir_file)),
+                None,
+                format!("{code_wire2api} move |task_callback| {code_call_inner_func_result}"),
+            ),
+            IrFuncMode::Stream { .. } => (
+                String::from("wrap::<_,_,_,()>"),
                 None,
                 format!("{code_wire2api} move |task_callback| {code_call_inner_func_result}"),
```

Comment:
nit: looks like the two branches has a bit of code duplication which may be removeable

Reply:
good point. I changed that